core/src/main/java/org/jruby/RubyDir.java

```diff
@@ -624,6 +624,15 @@ public IRubyObject rewind() {
         return this;
     }
 
+    @JRubyMethod(name = "empty?", meta = true)
+    public static IRubyObject empty_p(ThreadContext context, IRubyObject recv, IRubyObject arg) {
+        Ruby runtime = context.runtime;
+        RubyString path = StringSupport.checkEmbeddedNulls(runtime, RubyFile.get_path(context, arg));
+        RubyFileStat fileStat = runtime.newFileStat(path.asJavaString(), false);
+        boolean isDirectory = fileStat.directory_p().isTrue();
```

Comment:
This logic is taken from the `exist` method below. However, unlike that method, I want the exception to be thrown since `empty?` should throw an error if the path does not refer to an actual path. This is covered by the test `assert_raise(Errno::ENOENT) {Dir.empty?(@nodir)}`

Maybe there is a more expressive way to do this?